tests/recorder/recorder.go

```diff
@@ -0,0 +1,233 @@
+package main
```

Comment:
Yeah, I thought about it.  I do make some assumptions about what is being recorded (ie `go run` is hardcoded) so some abstractions need to be made.  Now the hard part, what to call it.  I am leaning towards `oscillo` (as in the scope for tracing), or we could be puny and call it `goscillo` which seems a bit obnoxious :)

Btw, if you need basic cli testing and do need to deal with fancy interactive prompts, then I have a useful project I use for testing most of my cli projects: https://github.com/coryb/osht, it makes testing via bash very easy.